pkg/cable/libreswan/libreswan_test.go

```diff
@@ -801,3 +827,166 @@ func (t *testDriver) assertNoActiveConnection(natInfo *natdiscovery.NATEndpointI
+				if fakecommand.CmdMatches(cmd, ContainSubstring("certutil"), "-N", "-d", "sql:/var/lib/ipsec/nss", "--empty-password") {
+					return fakecommand.InterceptorFuncs{Run: func() error {
+						return errors.New("database init failed")
+					}}
+				}
+
+				return fakecommand.InterceptorFuncs{}
+			})
+
+			controller := libreswan.NewCertificateHandler("test-cluster")
+			Expect(controller).NotTo(BeNil())
+
+			certData := map[string][]byte{
+				"ca.crt":  []byte("-----BEGIN CERTIFICATE-----\nMOCK_CA_CERT\n-----END CERTIFICATE-----"),
+				"tls.crt": []byte("-----BEGIN CERTIFICATE-----\nMOCK_CLIENT_CERT\n-----END CERTIFICATE-----"),
+				"tls.key": []byte("-----BEGIN PRIVATE KEY-----\nMOCK_CLIENT_KEY\n-----END PRIVATE KEY-----"),
+			}
+
+			err := controller.OnSignedCallback(certData)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("failed to initialize NSS database"))
+		})
+
+		It("should handle certificate loading failure", func() {
+			cmdExecutor = fakecommand.NewWithInterceptor(func(cmd *exec.Cmd) fakecommand.InterceptorFuncs {
+				if fakecommand.CmdMatches(cmd, ContainSubstring("certutil"), "-A", "-d", "sql:/var/lib/ipsec/nss",
+					"-n", "ca-cert", "-t", "C,,", "-i", "-", "-a") {
+					return fakecommand.InterceptorFuncs{Run: func() error {
+						return errors.New("certificate load failed")
+					}}
+				}
+
+				return fakecommand.InterceptorFuncs{}
+			})
+
+			controller := libreswan.NewCertificateHandler("test-cluster")
+			Expect(controller).NotTo(BeNil())
+
+			certData := map[string][]byte{
+				"ca.crt":  []byte("-----BEGIN CERTIFICATE-----\nMOCK_CA_CERT\n-----END CERTIFICATE-----"),
+				"tls.crt": []byte("-----BEGIN CERTIFICATE-----\nMOCK_CLIENT_CERT\n-----END CERTIFICATE-----"),
+				"tls.key": []byte("-----BEGIN PRIVATE KEY-----\nMOCK_CLIENT_KEY\n-----END PRIVATE KEY-----"),
+			}
+
+			err := controller.OnSignedCallback(certData)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("failed to load certificates into NSS database"))
+		})
+	})
+
+	Context("Certificate connection", func() {
+		When("connecting and disconnecting in certificate mode", func() {
+			It("should create and remove certificate-based connection", func() {
+				os.Setenv("CE_IPSEC_AUTHMODE", "cert")
+				defer os.Unsetenv("CE_IPSEC_AUTHMODE")
+
+				Expect(t.driver.Init()).To(Succeed())
+
+				natInfo := &natdiscovery.NATEndpointInfo{
+					Endpoint: subv1.Endpoint{
+						Spec: subv1.EndpointSpec{
+							ClusterID:  "remote",
+							CableName:  "submariner-cable-remote-192-68-2-1",
+							PrivateIPs: []string{"192.68.2.1"},
+							Subnets:    []string{"20.0.0.0/16"},
+						},
+					},
+					UseIP:     "172.93.2.1",
+					UseFamily: k8snet.IPv4,
+				}
+
+				_, err := t.driver.ConnectToEndpoint(natInfo)
+				Expect(err).To(Succeed())
+
+				err = t.driver.DisconnectFromEndpoint(&types.SubmarinerEndpoint{Spec: natInfo.Endpoint.Spec}, k8snet.IPv4)
+				Expect(err).To(Succeed())
```

Comment:
This test doesn't really verify anything. This is related to comment https://github.com/submariner-io/submariner/pull/3651/files#r2400777193 above.